src/pages/sourcecast/Sourcecast.tsx

```diff
@@ -107,202 +110,265 @@ export type StateProps = {
   uid: string | null;
 };
 
-class Sourcecast extends React.Component<SourcecastProps> {
-  public componentDidMount() {
-    this.props.handleFetchSourcecastIndex();
-  }
+const Sourcecast: React.FC<SourcecastProps> = props => {
+  const isMobileBreakpoint = useMediaQuery({ maxWidth: 768 });
```

Comment:
You may want to extract out the magic constant `768` instead of copy-pasting it everywhere.